### ISSUE TITLE
Not work the webhook under a proxy.

### ISSUE THREAD

**Issue by tomomura** [gitbucket/gitbucket#710]
gitbucket running on my server under a proxy. 
webhook work nice when destination ip on local. but not work when global ip. 
Do you gitbucket webhook supported under  a proxy environment?

**Comment by kerphi** [comment]
Did you try to run gitbucket with these -D options ?
```
-Dhttp.proxyHost=10.0.0.100 -Dhttp.proxyPort=8800 -Dhttps.proxyHost=10.0.0.100 -Dhttps.proxyPort=8800 -Djava.net.useSystemProxies=true
```
(replace 10.0.0.100 by your http proxy host and 8800 by your http proxy port)

Full command would be:

```
java -Dhttp.proxyHost=10.0.0.100 -Dhttp.proxyPort=8800 \
     -Dhttps.proxyHost=10.0.0.100 -Dhttps.proxyPort=8800 \
     -Djava.net.useSystemProxies=true \
     -jar gitbucket.war
```

**Comment by ecirtap** [comment]
It seems that the HttpClient library used in gitbucket/src/main/scala/gitbucket/core/service/WebHookService.scala does not take out of the box  the usual JVM defines related to the proxy management (see .https://hc.apache.org/httpcomponents-client-ga/tutorial/html/connmgmt.html#d5e485)

**Comment by nazoking** [comment]
fixed by #1203
